spec/features/finance/npq/course_payment_breakdown_spec.rb

```diff
@@ -53,27 +54,19 @@ def create_accepted_application(user, npq_course, npq_lead_provider)
   end
 
   def create_started_declarations(npq_application)
-    RecordDeclarations::Started::NPQ.call(
-      params: {
-        participant_id: npq_application.user.id,
-        course_identifier: npq_application.npq_course.identifier,
-        declaration_date: (npq_application.profile.schedule.milestones.first.start_date + 1.day).rfc3339,
-        cpd_lead_provider: npq_application.npq_lead_provider.cpd_lead_provider,
-        declaration_type: RecordDeclarations::NPQ::STARTED,
-      },
-    )
-  end
-
-  def create_started_declarations(npq_application)
-    RecordDeclarations::Started::NPQ.call(
-      params: {
-        participant_id: npq_application.user.id,
-        course_identifier: npq_application.npq_course.identifier,
-        declaration_date: (npq_application.profile.schedule.milestones.first.start_date + 1.day).rfc3339,
-        cpd_lead_provider: npq_application.npq_lead_provider.cpd_lead_provider,
-        declaration_type: RecordDeclarations::NPQ::STARTED,
-      },
-    )
+    stamp = npq_application.profile.schedule.milestones.first.start_date + 1.day
+
+    Timecop.freeze(stamp) do
```

Comment:
Could you use `freeze_time` instead of `timecop` which has been deprecated for a few years now, in favour of rails built in `ActiveSupport::Testing::TimeHelpers`.

https://edgeapi.rubyonrails.org/classes/ActiveSupport/Testing/TimeHelpers.html#method-i-freeze_time

Reply:
I don't see a deprecation warning anywhere? Those time helpers have slightly different APIs though. It doesn't appear to have the functionality to freeze time to a specific point in time.